Update the development environment

mwouts opened the pull request.

mahendrapaipuri commented on docs/developing.md:
````diff
@@ -37,19 +40,19 @@ Tests are executed with `pytest`. You can run them in parallel with for instance
+Finally, you can build the package and install it with
 ```
-pip install .
+pip install -U build wheel
+python -m build
+pip install dist/jupytext-x.x.x-py3-none-any.whl
````
Just out of curiosity, why build a `wheel` and install it instead of directly doing a `pip install .`?

mwouts replied:
Well I am afraid it just tells that I know so little about wheels and tars etc. Before `jupytext==1.16` I would use that to test the extension because the editable install would not include the extension, but from your comment it seems that I can remove that part, so I'll remove it!

mahendrapaipuri replied:
Ahh, ok I see. Yes, now we build the extension _always_, so we can directly do editable installs!

mahendrapaipuri commented on docs/developing.md:
````diff
@@ -22,11 +22,14 @@ Most of Jupytext's code is written in Python. To develop the Python part of Jupy
-We use the [pre-commit](https://pre-commit.com) package to run pre-commit scripts like `black` and `flake8` on the code.
+Install the `jupytext` package in development mode with
+```
+pip install '.[dev]'
````
I guess you are missing a `editable` flag here? `pip install -e '.[dev]'`?

mwouts replied:
Oh yes you're right! Thanks

mahendrapaipuri reviewed (commented):
I guess we need to update the instructions for developing extension as well. [This part](https://github.com/mwouts/jupytext/blob/af66065b464d6712bfdc92fc7a6821d24464b4f4/jupyterlab/packages/jupyterlab-jupytext-extension/README.md?plain=1#L23-L35) is obselete. The command `jupyter labextension develop --overwrite .` does not work anymore due to the way we changed the organization of the repo. That is why I added a [utility script](https://github.com/mwouts/jupytext/pull/1154/commits/f224da18724849cef0505f50c7c8cd79a9597131) to make a symlink of extension in dev mode. The [docs of `jupyter-server-proxy`](https://github.com/jupyterhub/jupyter-server-proxy/blob/0c57b2734b3c74beaa7df33f607c1392d7908b18/CONTRIBUTING.md?plain=1#L106-L186) can be a good reference on updating the documentation of extension development.

mwouts commented:
> I guess we need to update the instructions for developing extension as well. [This part](https://github.com/mwouts/jupytext/blob/af66065b464d6712bfdc92fc7a6821d24464b4f4/jupyterlab/packages/jupyterlab-jupytext-extension/README.md?plain=1#L23-L35) is obselete. The command `jupyter labextension develop --overwrite .` does not work anymore due to the way we changed the organization of the repo. That is why I added a [utility script](https://github.com/mwouts/jupytext/pull/1154/commits/f224da18724849cef0505f50c7c8cd79a9597131) to make a symlink of extension in dev mode. The [docs of `jupyter-server-proxy`](https://github.com/jupyterhub/jupyter-server-proxy/blob/0c57b2734b3c74beaa7df33f607c1392d7908b18/CONTRIBUTING.md?plain=1#L106-L186) can be a good reference on updating the documentation of extension development.

Nice ! The symlink is super helpful and it does make the development workflow much easier. I think that I will replace the obsolete paragraph with just `pip install -e '.[dev]'` then.

Also I'll revert the change on `pandoc>=3.0` since it breaks the conda CI, I will look into that later on.
